app/routers/Index.ts

```diff
@@ -0,0 +1,67 @@
+import { Router, Request, Response, NextFunction } from 'express';
+import { Dependencies } from '../Types';
+import Boom from 'boom';
+import CategoryController from '../controllers/CategoryController';
+import PostController from '../controllers/PostController';
+import bodyValidator from '../middleware/validation.middleware';
+import CreatePostDto from '../dto/CreatePost';
+import CreateCategoryDto from '../dto/CreateCategory';
+import UpdateCategoryDto from '../dto/UpdateCategory';
+
+class Index {
+  router: Router;
+  private postController: PostController;
+  private bodyValidator: typeof bodyValidator;
+  private categoryController: CategoryController;
+
+  constructor(opts: Dependencies) {
+    this.postController = opts.postController;
+    this.categoryController = opts.categoryController;
+    this.bodyValidator = opts.bodyValidator;
+    this.router = Router();
+    this.init();
+  }
+
+  public rootPath(req: Request, res: Response, next: NextFunction) {
+    res.status(200).send({ message: 'Server is up and running' });
+  }
+
+  public notFound(req: Request, res: Response, next: NextFunction) {
+    next(Boom.notFound('Path not found'));
+  }
+  private routePosts() {
+    this.router.post(
+      '/api/v1/posts',
+      this.bodyValidator(CreatePostDto),
+      this.postController.create
+    );
+    this.router.get('/api/v1/posts', this.postController.getAll);
+    this.router
+      .route('/api/v1/posts/:id')
+      .get(this.postController.getOne)
```

Comment:
Can you also add validator to url params aka the `id` now can be whatever string. So you have to make new validator which validates url parameters :)